File: src/main/java/io/openfuture/openmessanger/web/controller/UserController.java
package io.openfuture.openmessanger.web.controller;

import java.util.Collection;

import org.springframework.security.core.Authentication;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import io.openfuture.openmessanger.repository.entity.User;
import io.openfuture.openmessanger.service.UserService;
import io.openfuture.openmessanger.web.request.user.UserDetailsRequest;
import io.openfuture.openmessanger.web.response.UserDetailsResponse;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/user")
public class UserController {

    private final UserService userService;

    @GetMapping("/recipients")
    public Collection<User> getAllRecipientBySender(@RequestParam("sender") final String senderUsername) {
        return userService.getAllRecipientsBySender(senderUsername);
    }

    @GetMapping("/all")
    public Collection<User> getAllRegisteredUsers() {
        return userService.getAllUsers();
    }

    @PostMapping("/userDetails")
    public UserDetailsResponse getUserDetails(@RequestBody UserDetailsRequest request,
                                              @AuthenticationPrincipal Object authentication) {
        log.info("{}", authentication);
        return userService.getUserDetails(request);
    }

}

File: src/main/kotlin/io/openfuture/openmessanger/web/controller/UserController.kt
package io.openfuture.openmessanger.web.controller

import io.openfuture.openmessanger.repository.entity.User
import io.openfuture.openmessanger.service.UserService
import io.openfuture.openmessanger.web.request.user.UserDetailsRequest
import io.openfuture.openmessanger.web.response.UserDetailsResponse
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/user")
class UserController(
    val userService: UserService
) {
    companion object {
        val log: Logger = LoggerFactory.getLogger(UserController::class.java)
    }

    @GetMapping("/recipients")
    fun getAllRecipientBySender(@RequestParam("sender") senderUsername: String?): Collection<User?>? {
        return userService.getAllRecipientsBySender(senderUsername)
    }

    @GetMapping("/all")
    fun allRegisteredUsers(): Collection<User?> {
        return userService.allUsers()
    }

    @PostMapping("/userDetails")
    fun getUserDetails(
        @RequestBody request: UserDetailsRequest,
        @AuthenticationPrincipal authentication: Any?
    ): UserDetailsResponse? {
        log.info("{}", authentication)
        return userService.getUserDetails(request)
    }
}